src/lib.rs

```diff
@@ -276,4 +277,284 @@ mod tests {
+
+        fn va_surface_attribute(&mut self) -> Self::DescriptorAttribute {
+            self.0.clone()
+        }
+    }
+
+    #[test]
+    // Ignore this test by default as it requires libva-compatible hardware.
+    #[ignore]
+    fn enc_h264_demo() {
+        // Based on `gst-launch-1.0 videotestsrc num-buffers=1 ! video/x-raw,width=64,height=64,format=NV12 ! vaapih264enc ! filesink location=frame.h264`
+        // Frame created using `gst-launch-1.0 videotestsrc num-buffers=1 ! video/x-raw,width=64,height=64,format=NV12 ! filesink location=src/test_frame.nv12`
+        let raw_frame_nv12 = include_bytes!("test_frame.nv12");
+        let aligned_size = (raw_frame_nv12.len() + 4096 - 1) & !(4096 - 1);
+        let layout = unsafe { std::alloc::Layout::from_size_align_unchecked(aligned_size, 4096) };
+        let raw_frame_nv12_mem = unsafe { std::alloc::alloc(layout) };
+
+        unsafe { raw_frame_nv12_mem.copy_from(raw_frame_nv12.as_ptr(), raw_frame_nv12.len()) };
+
+        let display = Display::open().unwrap();
+
+        let format = bindings::constants::VA_RT_FORMAT_YUV420;
+        let entrypoint = bindings::VAEntrypoint::VAEntrypointEncSliceLP;
+        let profile = bindings::VAProfile::VAProfileH264ConstrainedBaseline;
+        let width = 64u32;
+        let height = 64u32;
+
+        let mut attrs = vec![bindings::VAConfigAttrib {
+            type_: bindings::VAConfigAttribType::VAConfigAttribRTFormat,
+            value: 0,
+        }];
+
+        display
+            .get_config_attributes(profile, entrypoint, &mut attrs)
+            .unwrap();
+
+        let config = display.create_config(attrs, profile, entrypoint).unwrap();
+
+        let mut buffers = vec![raw_frame_nv12_mem];
+
+        let test_frame = TestFrame(bindings::_VASurfaceAttribExternalBuffers {
+            pixel_format: bindings::constants::VA_FOURCC_NV12,
+            width,
+            height,
+            data_size: aligned_size as u32,
+            num_planes: 2,
+            pitches: [
+                width, // Y
+                width, // UV,
+                0, 0, // Unused
+            ],
+            offsets: [
+                0,              // Y
+                width * height, // UV,
+                0,              // Unused
+                0,              // Unused
+            ],
+            buffers: buffers.as_mut_ptr() as *mut _,
+            num_buffers: buffers.len() as u32,
+            flags: 0,
+            private_data: std::ptr::null_mut(),
+        });
+
+        let mut surfaces = display
+            .create_surfaces(
+                format,
+                None,
+                width,
+                height,
+                Some(UsageHint::USAGE_HINT_ENCODER),
+                vec![test_frame],
+            )
+            .unwrap();
+
+        let context = display
+            .create_context(
+                &config,
+                width,
+                ((height + 15) / 16) * 16,
```

Comment:
For simplicity, can't we just use `height` since it is constant?

Reply:
Sure!